### PR TITLE
chore(installation): remove non-OLM installation doc; add link to OLM quickstart (RHIDP-2707) (RHIDP-3497)

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

chore(installation): remove doc about non-OLM installation method; add extra link to OLM quickstart at https://olm.operatorframework.io/docs/getting-started/ ([RHIDP-2707](https://issues.redhat.com/browse/RHIDP-2707)) ([RHIDP-3497](https://issues.redhat.com/browse/RHIDP-3497))

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
* https://issues.redhat.com/browse/RHIDP-2707
* https://issues.redhat.com/browse/RHIDP-3497

### How to test this PR?
N/A

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] Code produced is complete
- [ ] Code builds without errors
- [ ] Tests are covering the bugfix
- [ ] Relevant user documentation updated
- [ ] Relevant contributing documentation updated

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by themr0c — APPROVED** [review]

File: modules/installation/proc-rhdh-deploy-eks-operator.adoc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [id='proc-rhdh-deploy-eks-operator_{context}']
 = Installing {product-short} on {eks-short} with the Operator
 
-You can install the {product} Operator with or without the Operator Lifecycle Manager (OLM) framework.
+Installation of the {product} Operator requires the Operator Lifecycle Manager (OLM) framework.
```
Comment (hmanwani-rh) [inline modules/installation/proc-rhdh-deploy-eks-operator.adoc:7]:
```suggestion
The {product} Operator installation requires the Operator Lifecycle Manager (OLM) framework.
```

**Review by rm3l — COMMENTED** [review]
I think we should also update the instructions for AKS: https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html/installing_red_hat_developer_hub_on_microsoft_azure_kubernetes_service/proc-rhdh-deploy-aks-operator_title-install-rhdh-aks#proc-rhdh-deploy-aks-operator_title-install-rhdh-aks

Right now, the first step in this procedure is `Obtain the Red Hat Developer Hub Operator manifest file, named rhdh-operator-<VERSION>.yaml [...]` (with no information on how to obtain that manifest), then `Apply the modified Operator manifest to your Kubernetes cluster`. There is no mention of OLM there.

But the instructions to install the Operator with OLM should be quite similar on AKS and EKS (not sure if the last step on changing the Service Type to `NodePort` is needed on AKS).

cc @gazarenkov

**Review by rm3l — APPROVED** [review]
> I think we should also update the instructions for AKS: [docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html/installing_red_hat_developer_hub_on_microsoft_azure_kubernetes_service/proc-rhdh-deploy-aks-operator_title-install-rhdh-aks#proc-rhdh-deploy-aks-operator_title-install-rhdh-aks](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html/installing_red_hat_developer_hub_on_microsoft_azure_kubernetes_service/proc-rhdh-deploy-aks-operator_title-install-rhdh-aks#proc-rhdh-deploy-aks-operator_title-install-rhdh-aks)
> 
> Right now, the first step in this procedure is `Obtain the Red Hat Developer Hub Operator manifest file, named rhdh-operator-<VERSION>.yaml [...]` (with no information on how to obtain that manifest), then `Apply the modified Operator manifest to your Kubernetes cluster`. There is no mention of OLM there.
> 
> But the instructions to install the Operator with OLM should be quite similar on AKS and EKS (not sure if the last step on changing the Service Type to `NodePort` is needed on AKS).
> 

As discussed, this can be updated in a follow-up PR to avoid blocking this one.

/unhold
/lgtm

cc @nickboldt

File: modules/installation/proc-rhdh-deploy-eks-operator.adoc [hunk 2]
```diff
@@ -18,7 +18,7 @@ You can install the {product-short} Operator on {eks-short} using the https://ol
 * You have set the context to the {eks-short} cluster in your current `kubeconfig`. For more information, see https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html[Creating or updating a kubeconfig file for an Amazon {eks-short} cluster].
 * You have installed `kubectl`. For more information, see https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html[Installing or updating kubectl].
 * You have subscribed to `registry.redhat.io`. For more information, see https://access.redhat.com/RegistryAuthentication[{company-name} Container Registry Authentication].
-* You have installed the Operator Lifecycle Manager (OLM). For more information about installation and troubleshooting, see https://operatorhub.io/how-to-install-an-operator#How-do-I-get-Operator-Lifecycle-Manager?[How do I get Operator Lifecycle Manager?]
+* You have installed the Operator Lifecycle Manager (OLM). For more information about installation and troubleshooting, see https://olm.operatorframework.io/docs/getting-started/[OLM QuickStart] or https://operatorhub.io/how-to-install-an-operator#How-do-I-get-Operator-Lifecycle-Manager?[How do I get Operator Lifecycle Manager?]
```
Comment (hmanwani-rh) [inline modules/installation/proc-rhdh-deploy-eks-operator.adoc:21]: @nickboldt According to the [RHIDP-2707](https://issues.redhat.com/browse/RHIDP-2707) description, "the operator install section doesn't mention that on Kubernetes clusters, you cannot simply discover the RHDH operator." Should we consider adding a note about Kubernetes clusters in this section?

Comment (rm3l) [reply modules/installation/proc-rhdh-deploy-eks-operator.adoc:21]: Yes, once OLM is installed and the RH CatalogSource is added, I think we can list all the available operators from that source and easily discover the RHDH one with this command:
```sh
kubectl get packagemanifests -n ${CatalogSourceNamespace}
```

For example:
```sh
$ k get packagemanifests -n rhdh-operator | grep rhdh      
rhdh                                           Red Hat Operators     8m21s
```

@hmanwani-rh Do you want this to be added to this page as well?

IMO, this is quite "simple" enough to discover the RHDH operator on non-OCP clusters, even without a UI. But do we need to mention this here? IMO, this is documenting how to install the operator, so we already know what to expect (like the name of the CatalogSource, namespace, ...).
That said, verifying that the RHDH operator shows up in the PackageManifests might still be helpful for troubleshooting.

**Comment by rm3l** [comment]
> I think we should also update the instructions for AKS: [docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html/installing_red_hat_developer_hub_on_microsoft_azure_kubernetes_service/proc-rhdh-deploy-aks-operator_title-install-rhdh-aks#proc-rhdh-deploy-aks-operator_title-install-rhdh-aks](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html/installing_red_hat_developer_hub_on_microsoft_azure_kubernetes_service/proc-rhdh-deploy-aks-operator_title-install-rhdh-aks#proc-rhdh-deploy-aks-operator_title-install-rhdh-aks)
> Right now, the first step in this procedure is `Obtain the Red Hat Developer Hub Operator manifest file, named rhdh-operator-<VERSION>.yaml [...]` (with no information on how to obtain that manifest), then `Apply the modified Operator manifest to your Kubernetes cluster`. There is no mention of OLM there.
> But the instructions to install the Operator with OLM should be quite similar on AKS and EKS (not sure if the last step on changing the Service Type to `NodePort` is needed on AKS).
> 

FYI, I've created a follow-up JIRA to track this: https://issues.redhat.com/browse/RHIDP-5032
I tried to update the instructions for AKS to require OLM. But since the OLM steps are quite similar, I think it might be worth refactoring and sharing the same OLM-related content for AKS and EKS, to avoid having a lot of duplicate content.

@hmanwani-rh @nickboldt Here is a quick draft PR (building on top of this one): https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/729

LMK your thoughts if it makes sense. If not, no problem to close it, and let the docs team work on https://issues.redhat.com/browse/RHIDP-5032 later.

**Pull request merged**